pkg/sql/region_util.go

```diff
@@ -1984,6 +2058,28 @@ func (p *planner) validateZoneConfigForMultiRegionTable(
 		expectedZoneConfig.NumReplicas = proto.Int32(0)
 	}
 
+	regionConfig := dbDesc.GetRegionConfig()
+	var primaryPref []zonepb.LeasePreference
+	if regionConfig != nil {
+		if regionConfig.SecondaryRegion != "" {
+			switch {
+			case desc.IsLocalityRegionalByTable():
+				rbt := desc.GetLocalityConfig().GetRegionalByTable()
+				if rbt.Region != nil {
+					primaryPref = synthesizeLeasePreferences(*rbt.Region)
+				} else {
+					primaryPref = synthesizeLeasePreferences(regionConfig.PrimaryRegion)
+				}
+			default:
+				primaryPref = synthesizeLeasePreferences(regionConfig.PrimaryRegion)
+			}
+
+			secondaryPref := synthesizeLeasePreferences(regionConfig.SecondaryRegion)
```

Comment:
Does it make sense to update `synthesizeLeasePreferences` to take both primary and secondary instead of doing the append below?